src/transaction/mvcc_table.cpp

```diff
@@ -707,17 +707,35 @@ mvcctable::get_global_oldest_visible () const
 }
 
 MVCCID
-mvcctable::update_global_oldest_visible ()
+mvcctable::update_global_oldest_visible (const MVCCID pts_oldest_visible /* = MVCCID_LAST */)
 {
-  if (m_ov_lock_count == 0)
+  if (pts_oldest_visible != MVCCID_ALL_VISIBLE)
     {
-      MVCCID oldest_visible = compute_oldest_visible_mvccid ();
       if (m_ov_lock_count == 0)
 	{
-	  assert (m_oldest_visible.load () <= oldest_visible);
-	  m_oldest_visible.store (oldest_visible);
+	  MVCCID ats_oldest_visible = compute_oldest_visible_mvccid ();
+	  if (m_ov_lock_count == 0)
+	    {
+	      /*
+	       * The assert below must be met to confirm there is no desynchronizaition.
+	       * But, it's allowed for now.
+	       * It is going to be dealt with soon in http://jira.cubrid.org/browse/LETS-563.
+	       */
+	      // assert (m_oldest_visible.load () <= pts_oldest_visible);
+	      assert (m_oldest_visible.load () <= ats_oldest_visible);
+	      if (ats_oldest_visible < pts_oldest_visible)
+		{
+		  m_oldest_visible.store (ats_oldest_visible);
+		}
+	      else
+		{
+		  m_oldest_visible.store (pts_oldest_visible);
```

Comment:
Should we assert that the value never goes backward? Or should this happen inherently by design?

Reply:
No, it must not. The asserts above check that, but one is commented out for now because of [LETS-563](http://jira.cubrid.org/browse/LETS-563). The assert is easy to fail for a while when a new PTS is connected.

Reply:
Right, I should have noticed.